src/transformers/generation/logits_process.py

```diff
@@ -474,12 +476,43 @@ def _calc_banned_ngram_tokens(
 
 class NoRepeatNGramLogitsProcessor(LogitsProcessor):
     r"""
-    [`LogitsProcessor`] that enforces no repetition of n-grams. See
+    N-grams are groups of "n" consecutive words, characters, or tokens taken from a sequence of text. Given a sentence:
+    " She runs fast ", the bi-grams (n = 2) would be ("she","runs") and ("runs","fast"). In text generation, avoiding
+    repetitions of word sequences provides a more diverse and coherent output. This [`LogitsProcessor`] enforces no
+    repetition of n-grams. See
     [Fairseq](https://github.com/pytorch/fairseq/blob/a07cb6f40480928c9e0548b737aadd36ee66ac76/fairseq/sequence_generator.py#L345).
+    It calls the <_calc_banned_ngram_tokens> function to determine the banned tokens (n-grams) based on the current
+    state of the generated sequences and the specified ngram_size. For each hypothesis, it sets the scores of banned
+    tokens to negative infinity (-inf), effectively discouraging the model from generating those tokens.
+
+    <Note> Use n-gram penalties with care. For instance, penalizing 2-grams (bigrams) in an article about the city of
+    New York might lead to undesirable outcomes where the city's name appears only once in the entire text.
+    [Reference](https://huggingface.co/blog/how-to-generate)
```

Comment:
This should be a `<Tip>` block. See [here](https://github.com/huggingface/transformers/blob/4a564490e149620282b9e946027e851f5f93aeae/src/transformers/generation/logits_process.py#L178) for an example